packages/core/src/test/shared/lsp/lspResolver.test.ts

```diff
@@ -0,0 +1,205 @@
+import { LanguageServerLocation } from '../../../shared/telemetry'
+
+const serverVersion = '1.0.0'
+const assetDirectory = 'path/to/assets'
+const serverName = 'myLS'
+
+/**
+ * Helper function for generating valid manifest for tests.
+ * @param platform
+ * @param arch
+ * @returns
+ */
+function manifestTarget(platform: 'linux' | 'darwin' | 'windows', arch: 'x64' | 'arm64'): Target {
+    return {
+        platform,
+        arch,
+        contents: [
+            {
+                filename: `${serverName}-${platform}-${arch}.zip`,
+                url: `https://example.com/lsp-${platform}-${arch}.zip`,
+                hashes: ['sha384:thisisahash'],
+                bytes: 100,
+            },
+            {
+                filename: `node-${platform}-${arch}`,
+                url: `https://example.com/temp-assets/node-${platform}-${arch}`,
+                hashes: ['sha384:thisisanotherhash'],
+                bytes: 200,
+            },
+        ],
+    }
+}
+/**
+ * Helper function for generating valid LspResult for tests.
+ * @param location
+ * @returns
+ */
+function lspResult(location: LanguageServerLocation): LspResult {
+    return {
+        location,
+        version: serverVersion,
+        assetDirectory: assetDirectory,
+    }
+}
+
+describe('lspResolver', function () {
+    let remoteStub: sinon.SinonStub
+    let localStub: sinon.SinonStub
+    let fallbackStub: sinon.SinonStub
+    let manifest: Manifest
+    let versionRange: Range
+    let isMac: boolean // TODO: remove this check once LSPResolver supports other OS.
+
+    before(function () {
+        isMac = process.platform === 'darwin'
+        remoteStub = sinon.stub(LanguageServerResolver.prototype, 'fetchRemoteServer' as any)
+        localStub = sinon.stub(LanguageServerResolver.prototype, 'getLocalServer' as any)
+        fallbackStub = sinon.stub(LanguageServerResolver.prototype, 'getFallbackServer' as any)
+        manifest = {
+            manifestSchemaVersion: '2.0.0',
+            artifactId: 'artifact-id',
+            artifactDescription: 'artifact-description',
+            isManifestDeprecated: false,
+            versions: [
+                {
+                    serverVersion: serverVersion,
+                    isDelisted: false,
+                    targets: [
+                        manifestTarget('linux', 'x64'),
+                        manifestTarget('linux', 'arm64'),
+                        manifestTarget('darwin', 'x64'),
+                        manifestTarget('darwin', 'arm64'),
+                        manifestTarget('windows', 'x64'),
+                    ],
+                },
+            ],
+        }
+        versionRange = new Range(`>=${serverVersion}`)
+    })
+
+    after(function () {
+        sinon.restore()
+    })
+
+    it('tries local cache first', async function () {
+        if (!isMac) {
+            this.skip()
+        }
+        localStub.resolves(lspResult('cache'))
```

Comment:
there are technically non stubbed versions of these tests in the integ or e2e folder btw. It might make sense to just add telemetry assertions there

Reply:
Where are these tests? I am having some trouble finding them.

Reply:
https://github.com/aws/aws-toolkit-vscode/blob/feature/amazonqLSP/packages/amazonq/test/e2e/lsp/lspInstaller.test.ts#L73

Reply:
What functionality is covered in this one thats not in those other ones?